### ISSUE TITLE
2.x: FlowableConcatMapEagerTest > testAsynchronousRun hung

### ISSUE THREAD

**Issue by akarnokd** [ReactiveX/RxJava#4399]
Reason unknown.

**Comment by akarnokd** [comment]
```
io.reactivex.internal.operators.flowable.FlowableConcatMapEagerTest > testAsynchronousRun FAILED
    java.lang.AssertionError: Value counts differ; Expected: 2000, Actual: 704 (latch = 1, values = 704, errors = 0, completions = 0)
        at io.reactivex.subscribers.TestSubscriber.fail(TestSubscriber.java:427)
        at io.reactivex.subscribers.TestSubscriber.assertValueCount(TestSubscriber.java:574)
        at io.reactivex.internal.operators.flowable.FlowableConcatMapEagerTest.testAsynchronousRun(FlowableConcatMapEagerTest.java:583)
```

**Comment by akarnokd** [comment]
Fixed via #4468. The issue was due to the possibility the inner queue was not ready when the switch to the next source happened and running into NPE or other kind of hangs. The inner queue is now volatile and checked for nullness before accessing it.
